clusterloader2/state/namespaces.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// InstancesState represents state of object replicas.
+type InstancesState struct {
+	Number int32
+	Object api.Object
+}
+
+// NamespaceState represents state of a single namespace.
+type NamespaceState map[string]InstancesState
```

Comment:
You shouldn't store a type itself in the map - it should store a pointer.

Reply:
pointers added